src/session.ts

```diff
@@ -96,43 +96,78 @@ async function withAuth(request: Request, { ensureSignedIn = false, debug = fals
         },
       });
     }
-    return { user: null };
+
+    return json(
+      {
+        user: null,
+        ...data,
+      },
+      init,
+    );
   }
 
-  const { sid: sessionId, org_id: organizationId, role, permissions } = decodeJwt<AccessToken>(session.accessToken);
+  const { sessionId, organizationId, role, permissions } = getClaimsFromAccessToken(session.accessToken);
 
-  return {
-    sessionId,
-    user: session.user,
-    organizationId,
-    role,
-    permissions,
-    impersonator: session.impersonator,
-    accessToken: session.accessToken,
-  };
+  return json(
+    {
+      sessionId,
+      user: session.user,
+      organizationId,
+      role,
+      permissions,
+      impersonator: session.impersonator,
+      accessToken: session.accessToken,
+      ...data,
+    },
+    {
+      status: init.status ?? 200,
+      headers: {
+        ...init.headers,
```

Comment:
Just realized that with `ResponseInit`, headers will be `HeadersInit` and this would be broken if the user doesn't pass a plain object 😢  (my bad here!)

Probably simplest just to type it directly and restrict the `headers` to an object since there are only three properties anyway.

```
interface InternalHeadersInit {
  status?: string;
  statusText?: string;
  headers?: Record<string, string>;
}
```

Alternatively, the [`merge-headers`](https://www.npmjs.com/package/merge-headers) package is quite small and simple. Probably no harm in using it or copying the implementation.